pkg/awx/connection.go

```diff
@@ -178,7 +188,7 @@ func (c *Connection) Close() {
 // new onw to the server.
 //
 func (c *Connection) ensureToken() error {
-	if c.token != "" {
+	if c.token != "" || c.bearer != "" {
```

Comment:
I sometimes see the test `len(x) != 0` and sometime `!= ""`.
Is there a difference? do we want to follow a standard?

Reply:
`len()` is more general (tests for nil as well) for string it's the same. I don't have a preference

Reply:
I'll change the other occurrence in this PR to be != as well